samples/extensions/open_cl_interop/CMakeLists.txt

```diff
@@ -0,0 +1,42 @@
+# Copyright (c) 2021, Arm Limited and Contributors
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 the "License";
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+ 
+# The OpenCL interoperability only works on Android
+get_filename_component(FOLDER_NAME ${CMAKE_CURRENT_LIST_DIR} NAME)
+get_filename_component(PARENT_DIR ${CMAKE_CURRENT_LIST_DIR} PATH)
+get_filename_component(CATEGORY_NAME ${PARENT_DIR} NAME)
+
+# OpenCL
+add_library(opencl INTERFACE)
+set(OPENCL_INCLUDE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/third_party/OpenCL-Headers/)
+target_sources(opencl INTERFACE ${OPENCL_INCLUDE_DIR}/CL/opencl.h)
+target_include_directories(opencl INTERFACE ${OPENCL_INCLUDE_DIR})
+
+if(ANDROID)
+    add_sample_with_tags(
+        ID ${FOLDER_NAME}
+        CATEGORY ${CATEGORY_NAME}
+        AUTHOR "Arm"
+        NAME "OpenCL Interoperability"
+        DESCRIPTION "Example showing sharing resources between OpenCL and Vulkan"
+        TAGS "arm"
+        LIBS opencl
+        FILES
+            open_cl_functions.inl
+            open_cl_utils.h
+            open_cl_utils.cpp)

```

Comment:
Is this the complete list of files that we want the compiler to compile?  How does open_cl_interop.cpp get compiled?

Reply:
It is included automatically. You can find the following lines in `bldsys/cmake/sample_helper.cmake`:
```
set(SRC_FILES
    ${TARGET_ID}.h
    ${TARGET_ID}.cpp
)

# Append extra files if present
if (TARGET_FILES)
    list(APPEND SRC_FILES ${TARGET_FILES})
endif()
```